sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClientBuilder.java

```diff
@@ -0,0 +1,152 @@
+     *
+     * @param endpoint The endpoint to connect to.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder endpoint(String endpoint) {
+        Objects.requireNonNull(endpoint);
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(ex));
+        }
+        clientImplBuilder.instanceUrl(endpoint);
+        return this;
+    }
+
+    /**
+     * Sets The HTTP pipeline to send requests through.
+     *
+     * @param pipeline the pipeline value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder pipeline(HttpPipeline pipeline) {
+        Objects.requireNonNull(pipeline);
+        clientImplBuilder.pipeline(pipeline);
+        return this;
+    }
+
+    /**
+     * Sets The serializer to serialize an object into a string.
+     *
+     * @param serializerAdapter the serializerAdapter value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder serializerAdapter(SerializerAdapter serializerAdapter) {
+        Objects.requireNonNull(serializerAdapter);
+        clientImplBuilder.serializerAdapter(serializerAdapter);
+        return this;
+    }
+
+    /**
+     * Sets The HTTP client used to send the request.
+     *
+     * @param httpClient the httpClient value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder httpClient(HttpClient httpClient) {
+        clientImplBuilder.httpClient(httpClient);
+        return this;
+    }
+
+    /**
+     * Sets The configuration store that is used during construction of the service client.
+     *
+     * @param configuration the configuration value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder configuration(Configuration configuration) {
+        clientImplBuilder.configuration(configuration);
+        return this;
+    }
+
+    /**
+     * Sets The logging configuration for HTTP requests and responses.
+     *
+     * @param httpLogOptions the httpLogOptions value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder httpLogOptions(HttpLogOptions httpLogOptions) {
+        clientImplBuilder.httpLogOptions(httpLogOptions);
+        return this;
+    }
+
+    /**
+     * Sets The retry policy that will attempt to retry failed requests, if applicable.
+     *
+     * @param retryPolicy the retryPolicy value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        clientImplBuilder.retryPolicy(retryPolicy);
+        return this;
+    }
+    /**
+     * Adds a custom Http pipeline policy.
+     *
+     * @param customPolicy The custom Http pipeline policy to add.
```

Comment:
You would also need to add service version for future.

Reply:
Just realized I missed that, added.